scripts/migrate_add_tag_base.py

```diff
@@ -1,24 +1,93 @@
-# Cortex migration script for migrating a cortex without tag:base values set.
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+'''
+synapse - migrate_add_tag_base.py
+Created on 6/1/17.
+
+Cortex migration script for migrating a cortex without tag:base values set.
+'''
+# Stdlib
+import argparse
+import logging
 import sys
+# Third Party Code
 import synapse.cortex as s_cortex
-url = sys.argv[1]
-print('Opening URL {}'.format(url))
-core = s_cortex.openurl(url)
-print('Getting tags')
-nodes = core.eval('syn:tag -syn:tag:base')
-print('Got {} tags'.format(len(nodes)))
-i = 0
-# Order the nodes.
-nodes.sort(key=lambda x: x[1].get('syn:tag'))
-with core.getCoreXact() as xact:
+
+logger = logging.getLogger(__name__)
+
+def upgrade_tagbase_xact(core, nodes):
+    '''
+    Wrap the upgrade of a local cortex in a transaction.
+    '''
+    with core.getCoreXact() as xact:
+        upgrade_tagbase_no_xact(core, nodes)
+
+def upgrade_tagbase_no_xact(core, nodes):
+    '''
+    Upgrade a set of tag nodes to have syn:tag:base properties.
+    '''
+    node_count = 0
     for node in nodes:
         tag = node[1].get('syn:tag')
+        if not tag:
+            continue
         if 'syn:tag:base' in node[1]:
-            print('Skipping: [{}]'.format(tag))
+            logger.debug('Skipping: [%s]', tag)
             continue
         parts = tag.split('.')
         base = parts[-1]
-        print('Setting syn:tag:base for [{}] to [{}]'.format(tag, base))
+        logger.debug('Setting syn:tag:base for [%s] to [%s]', tag, base)
         core.setTufoProps(node, base=base)
-        i += 1
-print('Migrated {} tag nodes.'.format(i))
+        node_count += 1
+    logger.debug('Migrated %s tag nodes.', node_count)
+
+def get_nodes_to_migrate(core):
+    logger.info('Getting tag nodes to migrate')
+    nodes = core.eval('syn:tag -syn:tag:base')
+    return nodes
+
+# noinspection PyMissingOrEmptyDocstring
+def main(options):  # pragma: no cover
+    if not options.verbose:
+        logging.disable(logging.DEBUG)
+
+    url = options.core
+    logger.info('Opening URL %s', url)
+    core = s_cortex.openurl(url)
+    nodes = get_nodes_to_migrate(core)
+    is_telepath = False
```

Comment:
lets add s_telepath.isproxy(item): that checks isinstance(synapse.telepath.Proxy) and use it here :D  This will probably be a common paradigm.